### PR TITLE
Nuke tweak subpackage

### PR DESCRIPTION
**Pull request by stevenschlansker**

### CODE REVIEW

**Comment by qualidafial** [comment]
While you're moving `Types`, should we rename it to `Generics`? I think Dropwizard has a class by that name for a similar purpose, it seems a more apt name, and doesn't conflict with `java.sql.Types`

File: core/src/main/java/org/jdbi/v3/mapper/CollectorFactory.java [hunk 1]
```diff
@@ -11,7 +11,7 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package org.jdbi.v3.tweak;
+package org.jdbi.v3.mapper;
```
Comment (qualidafial) [inline core/src/main/java/org/jdbi/v3/mapper/CollectorFactory.java]: Should the collector stuff go into its own package? Seems odd to put it in `mapper`.

File: core/src/main/java/org/jdbi/v3/statement/NoOpStatementRewriter.java [hunk 2]
```diff
@@ -11,12 +11,14 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package org.jdbi.v3;
+package org.jdbi.v3.statement;
```
Comment (qualidafial) [inline core/src/main/java/org/jdbi/v3/statement/NoOpStatementRewriter.java]: Can this go in the `rewriter` package instead?

**Comment by qualidafial** [comment]
`tweak/package.html` got moved into `statement` package, but the description doesn't fit now.

**Comment by qualidafial** [comment]
About halfway through review, I'm loving the hell out of this refactor.

**Comment by qualidafial** [comment]
Typo in `transaction/package.html`

File: core/src/main/java/org/jdbi/v3/util/EnumByNameMapperFactory.java [hunk 3]
```diff
@@ -13,14 +13,14 @@
  */
 package org.jdbi.v3.util;
```
Comment (qualidafial) [inline core/src/main/java/org/jdbi/v3/util/EnumByNameMapperFactory.java]: Let's move this to the `mapper` package.

File: core/src/main/java/org/jdbi/v3/util/EnumByOrdinalMapperFactory.java [hunk 4]
```diff
@@ -13,14 +13,14 @@
  */
 package org.jdbi.v3.util;
```
Comment (qualidafial) [inline core/src/main/java/org/jdbi/v3/util/EnumByOrdinalMapperFactory.java]: Let's move this to `mapper` package too

File: core/src/main/java/org/jdbi/v3/util/EnumMapper.java [hunk 5]
```diff
@@ -17,7 +17,7 @@
 import java.sql.SQLException;
 
 import org.jdbi.v3.StatementContext;
-import org.jdbi.v3.tweak.ColumnMapper;
+import org.jdbi.v3.mapper.ColumnMapper;
```
Comment (qualidafial) [inline core/src/main/java/org/jdbi/v3/util/EnumMapper.java]: Move to `mapper` package

File: core/src/test/java/org/jdbi/v3/GenericTypeTest.java [hunk 6]
```diff
@@ -16,6 +16,7 @@
 import static org.hamcrest.CoreMatchers.equalTo;
 import static org.hamcrest.MatcherAssert.assertThat;
 
+import org.jdbi.v3.util.GenericType;
```
Comment (qualidafial) [inline core/src/test/java/org/jdbi/v3/GenericTypeTest.java]: move test to `util` package

File: spring/src/main/java/org/jdbi/v3/spring/DBIFactoryBean.java [hunk 7]
```diff
@@ -23,7 +23,7 @@
 
 import org.jdbi.v3.Jdbi;
 import org.jdbi.v3.spi.JdbiPlugin;
-import org.jdbi.v3.tweak.StatementLocator;
+import org.jdbi.v3.statement.StatementLocator;
```
Comment (qualidafial) [inline spring/src/main/java/org/jdbi/v3/spring/DBIFactoryBean.java]: Boy am I glad I haven't gotten to the "remove all the old statement locator stuff" part of the `SqlLocator` task.. :)

**Comment by qualidafial** [comment]
Looks awesome. Just a few comments to address, then :shipit:

File: core/src/main/java/org/jdbi/v3/mapper/CollectorFactory.java (hunk 1 above)
Comment (stevenschlansker) [inline core/src/main/java/org/jdbi/v3/mapper/CollectorFactory.java]: I'm torn here.  We're getting a lot of packages, and Collectors are intimately tied to mapping and only have a few classes...

Comment (qualidafial) [inline core/src/main/java/org/jdbi/v3/mapper/CollectorFactory.java]: Yeah, kinda.

I guess it's fine there. Although I don't mind packages with only a few classes, personally.

**Comment by stevenschlansker** [comment]
@qualidafial the most interesting commit here is a5930b69 if you want to look at it

**Pull request merged**